### ISSUE TITLE
[RFC] Better skip links

### ISSUE THREAD

**Issue by jtoar** [redwoodjs/graphql#2338]
We more or less just shipped skip links but here's already some thoughts about how we could make them better. (Shout out to @Tobbe!) The goal generally is to make skip links more Redwoody. That's not super descriptive so here's some things about them that aren't so Redwoody ([here's the template for reference](https://github.com/redwoodjs/redwood/blob/main/packages/cli/src/commands/generate/layout/templates/layout.tsx.a11yTemplate)):
- [The css import is a little weird](https://github.com/redwoodjs/redwood/blob/6f3fd829dcb0df7864337e98110c4de94c033d6f/packages/cli/src/commands/generate/layout/templates/layout.tsx.a11yTemplate#L2). It'd be great to get rid of it, but maybe not hide it completely. One option is to add it to [index.css](https://github.com/redwoodjs/redwood/blob/main/packages/create-redwood-app/template/web/src/index.css). I'd imagine most people would want to style the skip link so we definitely shouldn't get in the way of that. Any other thoughts on how to do this are welcome
- The URL, by default, changes to #reach-skip-nav. Devs might wonder if that means something special, but it really doesn't; it's just the default. So maybe we should think about changing #reach-skip-nav to #redwood-skip-nav, or something more generic like #main-content. We could go about this just by generating the skip link components with default `contentId` and `id`, or wrap the component so that those are it's new defaults.

Apart from those, here's some other suggestions:
- [ ] Should we change this to the default layout generator? right now the default layout generator is just react fragments, so if we did we wouldn't be missing much. But we should be sure we got this component right if we do!
- [ ] Should we provide our own sorta monkey-patched `<Nav>` and `<Main>` components with skip links built? We might have to make use of React Portals here. The general theme of this suggestion is can we make things more automatic? Seems like we should be able to in some way.

**Comment by Tobbe** [comment]
> * [The css import is a little weird](https://github.com/redwoodjs/redwood/blob/6f3fd829dcb0df7864337e98110c4de94c033d6f/packages/cli/src/commands/generate/layout/templates/layout.tsx.a11yTemplate#L2). It'd be great to get rid of it, but maybe not hide it completely. One option is to add it to [index.css](https://github.com/redwoodjs/redwood/blob/main/packages/create-redwood-app/template/web/src/index.css). I'd imagine most people would want to style the skip link so we definitely shouldn't get in the way of that. Any other thoughts on how to do this are welcome

If we do add it to index.css (which I think is a good idea), I think we should add the actual styles, not just import/include the reach css. I agree that people probably want to style the skip link, and having the styles right there makes that easier.



> * So maybe we should think about changing #reach-skip-nav to #redwood-skip-nav, or something more generic like #main-content.

I mentioned #rw-skip-nav, but I actually like your suggestion of #main-content more 👍 



> * Should we change this to the default layout generator?

I vote we do! And if we do, I think we can look at this example from the [tutorial](https://learn.redwoodjs.com/docs/tutorial/layouts/#back-home-again). I think that's a pretty typical use of a layout, and we should make sure whatever we put in the layout by default is easily transformed to that.



> * Should we provide our own sorta monkey-patched `<Nav>` and `<Main>` components with skip links built? We might have to make use of React Portals here.

Another option, instead of portals, is to have `<Router>` include an empty container for the skip-link and have `<Nav>` put the link in there.
